pallets/gear/src/mock.rs

```diff
@@ -52,6 +56,62 @@ pub(crate) const USER_3: AccountId = 3;
 pub(crate) const LOW_BALANCE_USER: AccountId = 4;
 pub(crate) const BLOCK_AUTHOR: AccountId = 255;
 
+#[derive(Encode, Decode, Clone, Eq, PartialEq, Default, TypeInfo)]
+pub struct VoucherLegitimate;
+
+impl SignedExtension for VoucherLegitimate {
```

Comment:
The code looks similar to other testing mocks. Can it be refactored anyhow?

Reply:
Nope: due to inability to cast some specific `mock*::RuntimeCall` into ref to `pallet_gear_voucher::Call`